### PR TITLE
sink (ticdc): disable batch update dml when downstream is mysql

### LINKED ISSUE #8420: delete and update events may not be correctly sinked to downstream
### What did you do?

1. run `csv_storage_basic` test 
```
CASE="csv_storage_basic" docker-compose -f ./deployments/ticdc/docker-compose/docker-compose-storage-integration.yml up
```
2. run `sync_diff_inspector` to check difference between upstream and downstream.

### What did you expect to see?

All dml events can be sinked correctly.

### What did you see instead?

Delete and update events are not sinked properly.

### Versions of the cluster

TiCDC version (execute `cdc version`):

```console
v6.7.0-master
```

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #8420 

#### Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
Fix a bug that may causes data loss when downstrem is mysql.
```

### CODE REVIEW

File: cdc/sink/dmlsink/txn/mysql/mysql.go [hunk 1]
```diff
@@ -442,10 +442,20 @@ func (s *mysqlBackend) batchSingleTxnDmls(
...
-			sqls = append(sqls, s...)
-			values = append(values, v...)
+		if s.cfg.IsTiDB {
+			for _, rows := range updateRows {
+				s, v := s.genUpdateSQL(rows...)
+				sqls = append(sqls, s...)
+				values = append(values, v...)
+			}
+		} else {
+			for _, rows := range updateRows {
```
Comment (CharlesCheung96) [inline cdc/sink/dmlsink/txn/mysql/mysql.go:452]: Whether some comments need to be added here?

**Review by nongfushanquan — APPROVED** [review]

**Review by zhaoxinyu — APPROVED** [review]

**Pull request merged**